kedro/io/data_catalog.py

```diff
@@ -311,6 +360,28 @@ def _get_dataset(
 
         return data_set
 
+    def _get_resolved_dataset(
```

Comment:
I'd rename this to `_resolve_dataset` because that's what actually happens in this method. 
```suggestion
    def _resolve_dataset(
```

Reply:
Done